### ISSUE TITLE
Unet训练自己的数据集 跑完一个epoch就停了

### ISSUE THREAD

**Issue by YRDTSS** [PaddlePaddle/PaddleSeg#1521]
最近在学习飞桨，感觉飞桨是个很不错的框架。但我现在用Unet训练自己数据集的时候训练了10次模型就自己停了。
这是我的配置文件
_base_: '../_base_/haowei.yml'

batch_size: 1
iters: 300

train_dataset:
  type: Dataset
  dataset_root: data/haowei_seg
  train_path: data/haowei_seg/train.txt
  num_classes: 2
  transforms:
    - type: ResizeStepScaling
      min_scale_factor: 0.5
      max_scale_factor: 2.0
      scale_step_size: 0.25
    - type: RandomPaddingCrop
      crop_size: [640, 640]
    - type: RandomHorizontalFlip
    - type: RandomDistort
      brightness_range: 0.4
      contrast_range: 0.4
      saturation_range: 0.4
    - type: Normalize
  mode: train

val_dataset:
  type: Dataset
  dataset_root: data/haowei_seg
  val_path: data/haowei_seg/val.txt
  num_classes: 2
  transforms:
    - type: Normalize
  mode: val

optimizer:
  type: sgd
  momentum: 0.9
  weight_decay: 4.0e-5

lr_scheduler:
  type: PolynomialDecay
  learning_rate: 0.0025
  end_lr: 0
  power: 0.9

loss:
  types:
    - type: CrossEntropyLoss
  coef: [1]

model:
  type: UNet
  num_classes: 2
  use_deconv: False
  pretrained: Null

**Comment by YRDTSS** [comment]
我的数据集比较小，只有20张图像，不知道是否和这个有关系。希望有大佬看到能指点下

**Comment by YRDTSS** [comment]
训练是在windows10 上训练的，已经执行了官方给的那个测试程序，在output里也有结果图。训练的时候报了一些warning，如下
W1111 15:26:28.741739  3560 device_context.cc:404] Please NOTE: device: 0, GPU Compute Capability: 8.6, Driver API Version: 11.1, Runtime API Version: 11.2
W1111 15:26:28.741739  3560 device_context.cc:422] device: 0, cuDNN Version: 8.1.
W1111 15:26:28.741739  3560 device_context.cc:437] WARNING: device: 0. The installed Paddle is compiled with CUDA 11.2, but CUDA runtime version in your machine is 11.1, which may cause serious incompatible bug. Please recompile
or reinstall Paddle with compatible CUDA version.
C:\ProgramData\Anaconda3\envs\paddle_env\lib\site-packages\paddle\nn\layer\norm.py:641: UserWarning: When training, we now always track global mean and variance.
  "When training, we now always track global mean and variance.")
C:\ProgramData\Anaconda3\envs\paddle_env\lib\site-packages\paddle\fluid\dygraph\math_op_patch.py:239: UserWarning: The dtype of left and right variables are not the same, left dtype is paddle.float32, but right dtype is paddle.in
t64, the right dtype will convert to paddle.float32


有个警告是说cuda的版本的问题，我下载的paddle是11.2的 但是我的电脑是11.1的，这个有很大的影响吗？

**Comment by YRDTSS** [comment]
我刚刚使用一个人像分割的数据集，数据量肯定够了，模型第一次评估完就停下来了，我不知道是否哪个环节操作出问题了
![image](https://user-images.githubusercontent.com/54494631/141266534-3615a859-23be-4cf5-ba42-c085033247ac.png)

**Comment by YRDTSS** [comment]
我在一些问题中看到了和我一样的情况，我把transforms策略都去了，只留了一个Normalize。bachsize设成了1，loss换成了CrossEntropyLoss+LovaszSoftmaxLoss，但是任然没有效果。模型在第一次评估完之后就停止运行了，没有报错也没有其他的提示。有大佬看到麻烦指点下，谢谢了。

**Comment by YRDTSS** [comment]
helpme plz  T.T

**Comment by haoyuying** [comment]
您好，可以在aistudio上提供一个可复现的环境吗，方便我们排查原因，因为没有报错，很难定位问题。
现在可以给你一个排查方向，可以把训练过程和评估过程分开，看两者分别执行是否是正常的。

**Comment by YRDTSS** [comment]
好的 我在aistudio看下怎么弄一个复现的环境> 您好，可以在aistudio上提供一个可复现的环境吗，方便我们排查原因，因为没有报错，很难定位问题。
> 
> 现在可以给你一个排查方向，可以把训练过程和评估过程分开，看两者分别执行是否是正常的。
